### ISSUE TITLE
[BUG] DockingManager::FindEmptyContainer() possible buffer overflow

### ISSUE THREAD

**Issue by xomx** [notepad-plus-plus/notepad-plus-plus#15850]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

Due to the N++ Docking Manager design imperfection, there is a possibility of a buffer overflow in its `DockingManager::FindEmptyContainer()` method.

Erroneous memory overwriting in the above mentioned func usually does not lead to an immediate visible N++ crash, but if the user continues to use the undocking/docking of the N++ panels during the app life, it does lead to constant corruption of unpredictable parts of the app heap memory and consequently to the possible app-crash (or worse - to a possible data corruption).

This bug can also be intentionally exploited if an attacker manually changes the saved value of `prev="..."` to, say, `prev="99999999"`. Then the N++ would crash immediately upon any attempt to undock a panel.

### Steps To Reproduce

1.  Build a fresh Debug build of the N++ and launch it.
2.  Ctrl+F and then click `Find All in Current Document`
3.  Undock the `Search results` panel to the floating state
4.  Redock it back
5.  Close N++

Now in your relevant config.xml should be these Docking Manager data:
```
        <GUIConfig name="DockingManager" leftWidth="200" rightWidth="200" topHeight="200" bottomHeight="200">
            <FloatingWindow cont="4" x="183" y="344" width="878" height="624" />
            <PluginDlg pluginName="Notepad++::InternalFunction" id="0" curr="3" prev="4" isVisible="yes" />
            <ActiveTabs cont="0" activeTab="-1" />
            <ActiveTabs cont="1" activeTab="-1" />
            <ActiveTabs cont="2" activeTab="-1" />
            <ActiveTabs cont="3" activeTab="0" />
            <ActiveTabs cont="4" activeTab="-1" />
        </GUIConfig>
```

6.  Launch the N++ again and close it immediately.

Now your relevant config.xml data should be like:
```
        <GUIConfig name="DockingManager" leftWidth="200" rightWidth="200" topHeight="200" bottomHeight="200">
            <PluginDlg pluginName="Notepad++::InternalFunction" id="0" curr="3" prev="4" isVisible="yes" />
            <ActiveTabs cont="0" activeTab="-1" />
            <ActiveTabs cont="1" activeTab="-1" />
            <ActiveTabs cont="2" activeTab="-1" />
            <ActiveTabs cont="3" activeTab="-1" />
        </GUIConfig>
```

7. Set a breakpoint in MSVS on the `DockingManager::FindEmptyContainer()` method.
8. Launch a N++ debugging session.
9. Ctrl+F in N++ and then click `Find All in Current Document`
10. Try to undock the `Search results` panel to the floating state.
11. Now watch in debugger how the `pPrevDockList` buffer is being corrupted (see the pics below).
12. Buffer overflow heap manager msgbox warning, when you reach the `delete [] pPrevDockList;` line.

### Current Behavior

![npp-DockingManager-buffer-overflow-before](https://github.com/user-attachments/assets/cb875264-3276-4f10-b5d5-5a1de1b0d775)

![npp-DockingManager-buffer-overflow-after](https://github.com/user-attachments/assets/a21763d0-0a77-4283-8cde-0df222d9099b)

![npp-DockingManager-buffer-overflow-warning-msg](https://github.com/user-attachments/assets/678e5044-5a2e-492a-a5e5-a3d1384c1ec4)

### Expected Behavior

Such buffer overflow is not possible under any circumstances.

### Debug Information

```shell
Probably any N++ version (with the docking panels possibility).
```

**Comment by rdipardo** [comment]
Sounds familiar . . . https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13084#issuecomment-1783787209

**Comment by xomx** [comment]
@rdipardo 
Now I see that I probably already once met this bug before - https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14478#issuecomment-1887640464 :

"...also found a Docking Manager heap buffer overrun which can be connected to these issues )"

And there is probably also hidden the reason of this buggy situation:

"Now to the question - is it ok that the 'Search Results' panel behaves differently than the other N++ internal ones and does not reappear after N++ relaunch?"

So my STR with the SR-panel are probably the result of the decision to let the SR-panel behave differently - it has the `isVisible="yes"` set in its config.xml Docking Manager part but N++ ignores it (to not show an empty SR-panel at the N++ relaunch...).

But still - just like in the example you linked above, N++ should not allow any invalid data in the config.xml to crash it.

**Comment by donho** [comment]
@xomx 
So this PR fixes eventually #14478?
